inst/string.m

```diff
@@ -1276,3 +1284,31 @@ function disp (this)
     endif
   endfor
 endfunction
+
+## Test string constructor
+%!test
+%! str = string (["a";"b";"c"]);
+%! assert (cell (str), {"a";"b";"c"})
+%!test
+%! str = string ({"a";"b";"c"});
+%! assert (cell (str), {"a";"b";"c"})
+%!test
+%! str = string ({"a";"";"c"});
+%! tfM = ismissing (str);
+%! assert (cell (str), {"a";[];"c"})
+%! assert (tfM, logical ([0; 1; 0]))
+%!test
+%! str = string ([1 2 3 NaN 5]);
+%! tfM = ismissing (str);
+%! assert (cell (str), {"1", "2", "3", [], "5"})
```

Comment:
Similarly, I think in `cellstr(str)`, a `string(missing)` will/should convert to `{''}` instead of `{[]}`, so I think element 4 here should be `""`.